modules/sfp_stackoverflow.py

```diff
@@ -0,0 +1,255 @@
+
+        if matches:
+            for match in matches:
+                if self.sf.validIP6(match) and not(self.sf.isValidLocalOrLoopbackIP(match)):
+                    ips.add(match)
+            return list(ips)
+        else:
+            return None
+
+    def handleEvent(self, event):
+        eventData = event.data
+
+        if self.errorState:
+            return
+
+        if eventData in self.results:
+            self.debug(f"Skipping {eventData}, already checked.")
+            return
+
+        self.results[eventData] = True
+
+        query_results = self.query(eventData, "excerpts")
+        items = query_results.get('items')
+        allEmails = []
+        allUsernames = []
+        allIP4s = []
+        allIP6s = []
+
+        if items is None:
+            return
+        # Iterate through all results from query, creating raw_rir_data events and extracting emails
+        for item in items:
+            if self.checkForStop():
+                return
+
+            body = item["body"]
+            excerpt = item["excerpt"]
+            question = item["question_id"]
+            text = body + excerpt
+
+            # create raw_rir_data event
+            e = SpiderFootEvent('RAW_RIR_DATA',
+                                str("<SFURL>https://stackoverflow.com/questions/") + str(question) + str("</SFURL>") + str("\n") + str(item), self.__name__, event)
+            self.notifyListeners(e)
+
+            emails = self.sf.parseEmails(text)
+            if emails is not None:
+                for email in emails:
+                    allEmails.append(str(email))
+
+            questionId = item["question_id"]
+            username = self.extractUsername(questionId, event)
+            if username is not None:
+                allUsernames.append(username)
+
+            ip4s = self.extractIP4s(text)
+            if ip4s is not None:
+                allIP4s.append(ip4s)
+
+            ip6s = self.extractIP6s(text)
+            if ip6s is not None:
+                allIP6s.append(ip6s)
+
+        # create events for emails, username and IPs
+        for email in allEmails:
+            email = str(email)
+            if eventData in email:
+                e = SpiderFootEvent('EMAILADDR', email, self.__name__, event)
+                self.notifyListeners(e)
+            else:
+                e = SpiderFootEvent('AFFILIATE_EMAILADDR', email, self.__name__, event)
+                self.notifyListeners(e)
+
+        for username in allUsernames:
+            e = SpiderFootEvent('USERNAME', username, self.__name__, event)
```

Comment:
I'm still seeing non-username types for usernames. If you test against `tesla.com`, you'll see usernames like `A J E` and `stacj aa`. Probably anything with a space should be disregarded, but I'm also not sure how good the quality of usernames are here.

Reply:
All of that information is pulled directly from the displayname value in the API response:
![image](https://user-images.githubusercontent.com/42327230/148368617-ca7ce85e-b021-44b2-bab9-327b68e5475b.png)

Users on StackOverflow are allowed to set a username with spaces, "stacj aa" is somebodies actual username
https://stackoverflow.com/users/14085501/stacj-aa?tab=topactivity

If I remove anything with a space, we'll miss out on legitimate users

Reply:
SpiderFoot internally as well as the modules processing `USERNAME` events may encounter issues if spaces are contained. Maybe if it has a space, report it instead as a "Potential full name: <username>" `RAW_RIR_DATA` event for `sfp_names` to handle?

Reply:
Cool! I think that's should be sorted now - looks like the human names are all correctly being reported and all of the usernames don't have spaces and also aren't human names. 👍